src/main/java/org/motechproject/mots/service/AutomatedReportService.java

```diff
@@ -0,0 +1,51 @@
+package org.motechproject.mots.service;
+
+import java.util.List;
+import java.util.Objects;
+import org.motechproject.mots.domain.AutomatedReportSettings;
+import org.motechproject.mots.dto.AutomatedReportSettingsDto;
+import org.motechproject.mots.exception.AutomatedReportException;
+import org.motechproject.mots.mapper.AutomatedReportSettingsMapper;
+import org.motechproject.mots.repository.AutomatedReportSettingsRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AutomatedReportService {
+
+  @Autowired
+  private AutomatedReportSettingsRepository automatedReportSettingsRepository;
+
+  @Autowired
+  private SchedulerService schedulerService;
+
+  private static final AutomatedReportSettingsMapper MAPPER =
+      AutomatedReportSettingsMapper.INSTANCE;
+
+  public void updateSettings(AutomatedReportSettingsDto settingsDto) {
+    AutomatedReportSettings automatedReportSettings =
+        automatedReportSettingsRepository.findOneByJobName(settingsDto.getJobName());
+    if (automatedReportSettings == null) {
+      throw new AutomatedReportException("There is no job with such name");
+    } else {
+      if (isSettingDifferent(settingsDto, automatedReportSettings)) {
+        MAPPER.updateFromDto(settingsDto, automatedReportSettings);
+        automatedReportSettingsRepository.save(automatedReportSettings);
+        schedulerService.updateJob(settingsDto);
+      }
+    }
+  }
+
+  public List<AutomatedReportSettingsDto> getAll() {
+    return MAPPER.toDtos(automatedReportSettingsRepository.findAll());
+  }
+
+  private boolean isSettingDifferent(AutomatedReportSettingsDto settingsDto,
+      AutomatedReportSettings automatedReportSettings) {
+    return !Objects.equals(settingsDto.getIntervalInSeconds(),
```

Comment:
- when settingsDto job is disabled and automatedReportSettings is enabled you should remove the job
- when settingsDto job is enabled and automatedReportSettings is disabled you should add (reschedule) the job 
- when settingsDto job is enabled and startDate or interval was changed reschedule the job